src/components/ShortcutBox.vue

```diff
@@ -1,440 +1,412 @@
-          ...oldFilters,
-          newFilter,
-        ])
-      } else {
-        this.newFilter = filter
-      }
-    },
-    onChange(myFilterData) {
-      console.log('onChange()', myFilterData, this.select)
-      if (this.select) this.isEnterPressed = false
-      if (myFilterData.key === "default.search") {
-        this.submitSearchString()
-      }
-      else if (myFilterData?.isFilterLink) {
-        this.selectFilter(myFilterData)
-      }
-      else if (myFilterData?.value) {
-        url.pushNewFilters([
-          ...url.readFilters(this.$route),
-          myFilterData,
-        ])
-        this.clear()
-      }
-
-      setTimeout(() => { // no idea why this is necessary but it is
-        this.searchString = ""
-        this.select = null
-        this.suggestions = []
-      })
-    },
-    onEnterKeyup() {
-      if (!this.isEnterPressed) return
-      if (!this.searchString && this.showExamples) {
-        // we're on the landing page or something like it
-        url.pushToRoute(this.$router, {name: "Serp", params: {entityType: this.entityType}})
-        return
-      }
-
-      const filterKey = this.newFilter?.key ?? "default.search"
-      url.createFilter(this.entityType, filterKey, this.searchString)
-      this.isEnterPressed = false
-    },
-    submitSearchString() {
-      console.log("submitSearchString")
-      if (!this.searchString) {
-        url.pushToRoute(this.$router, {name: "Serp", params: {entityType: this.entityType}})
-      } else {
-        const searchFilter = createSimpleFilter(this.entityType, "default.search", this.searchString)
-        url.pushNewFilters([
-          ...url.readFilters(this.$route),
-          searchFilter
-        ])
-      }
-    },
-    viewWorks(id) {
-      console.log("view my works", id)
-
-      const entityType = entityTypeFromId(id)
-      if (!id || !entityType) return
-
-      const filter = createSimpleFilter(
-          "works",
-          entityConfigs[entityType].filterKey,
-          id,
-      )
-      url.pushNewFilters([
-        ...url.readFilters(this.$route),
-        filter,
-      ])
-      this.clear()
-
-    },
-    goToEntity(id) {
-      console.log("goToEntity()", id)
-      url.pushToRoute(this.$router, {
-        name: "EntityPage",
-        params: urlPartsFromId(id)
-      })
-
-    },
-    trySearch(str) {
-      setTimeout(() => {
-        this.searchString = str
-        console.log("trySearch", str)
-        this.$refs.shortcutBox.focus()
-      }, 100)
-    },
-
-    getSuggestions: _.debounce(async function () {
-      const fulltextSearchFilter = createSimpleFilter(this.entityType, "default.search", this.searchString)
-
-      // lol hack much?
-      if (this.searchString === "coriander OR cilantro") {
-        this.suggestions = [fulltextSearchFilter]
-        return
-      }
-
-      this.isLoading = true
+<script setup lang="ts">
+  import { ref,  useTemplateRef, computed, watch, onMounted, onBeforeUnmount, toRaw } from 'vue';
```

Comment:
here you can see an alternative syntax for the composition api and the [setup function](https://vuejs.org/api/sfc-script-setup). 

Using <script setup> means you’re exclusively working with the Composition API in that file.

Options API	Composition API (in <script setup>)
data	                 ref or reactive
methods	         Define regular functions
computed	 computed function
watch	          watch or watchEffect
mounted	          onMounted lifecycle hook
props	          Automatically exposed or defineProps
emit	                   defineEmits